version: "3.9"
services:
  db:
    container_name: ${CONTAINER_NAME}_db
    image: kartoza/postgis:14-3.3
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DBNAME=${POSTGRES_DBNAME}
    volumes:
      - ./docker/database:/var/lib/postgresql/data
    ports:
      - "${DB_PORT}:5432"
    restart: always
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${POSTGRES_USER}", "-h", "cw_db"]
      interval: 5s
      timeout: 3s
      retries: 5

  api:
    container_name: ${CONTAINER_NAME}_api
    build: ./docker/api
    command: sh -c "./run_api.sh"
    volumes:
      - ./api/:/app/api
      - ./run.py:/app/run.py
      - ./.env:/.env
    ports:
      - "${API_PORT}:5000"
    restart: always
    healthcheck:
      test:
        ["CMD", "curl", "-I", "--fail", "http://http://171.19.0.1:5000/status"]
      interval: 30s
      timeout: 3s
      retries: 5
